### ISSUE TITLE
Fix: Clean Architecture rules in Coroutine template 

### ISSUE THREAD

**Issue by doannimble** [nimblehq/android-templates#131]
## Issue

Our current implementation of Clean Architecture rules in Coroutine template is not correct, details : 
- Domain module : depends on Data module and Android framework
- Data module : no repository implementation, no mapper from Data to Domain model

## Expected

- Domain should contain only Kotlin code which has no dependency
- Data module should have Repository implementation, mapper from Data to Domain model

## Steps to reproduce

N/A

**Comment by lydiasama** [comment]
@doannimble I think it's not a bug. 🤔  Should it be a chore ticket for improvement?

**Comment by doannimble** [comment]
@lydiasama I don't think so. Which does not correct should be a bug. This is a correction, not an improvement.
Just a little bit improvement of Detekt included by using the latest version.

**Comment by sleepylee** [comment]
@doannimble @lydiasama: just answer a simple question: did the app work as expected before? If yes - it's a bug, otherwise it's not.
Do these changes alter the app behavior or just a refactor work at the codebase level? - if not, it's a refactor; if it changes the app behavior i.e performing better: enhancement

**Comment by luongvo** [comment]
@doannimble I'm quite sure this is just an improvement task, not a bug.

**Comment by doannimble** [comment]
@sleepylee @lydiasama 
- Did the app work as expected before ? : Yes
- Do these changes alter the app behavior or just a refactor work at the codebase level? : Refactor work at codebase level.
So we only catching bug on app level? If that is true, then I agree it a refactor or improvement tasks.

@luongvo I agree, it should be refactor or improvement.

**Comment by sleepylee** [comment]
@doannimble yes, and it could happens too; sometimes if there is an issue with our subscription model (missing context) it could crash the app when performing an action on the UI thread; or duplicated emission when we forgot to unsubscribe sth; so in that cases, we need a space to log the bug

**Comment by doannimble** [comment]
@sleepylee I got it. Thank you 🙏

**Comment by lydiasama** [comment]
@sleepylee Should we create content about how to categorize these as IBM did? 😅

> @doannimble @lydiasama: just answer a simple question: did the app work as expected before? If no - it's a bug, otherwise it's not.
> Do these changes alter the app behavior or just a refactor work at the codebase level? - if not, it's a refactor; if it changes the app behavior i.e performing better: enhancement.
> 
> Here I'm taking a reference definition from IBM:
> 
> https://www.ibm.com/topics/bug-tracking
> ![Screenshot_20210927-103651_Chrome.jpg](https://user-images.githubusercontent.com/13435717/134841993-cf5454dd-989a-457c-b18d-a3b0fcaddb6d.jpg)

**Comment by sleepylee** [comment]
@lydiasama I thought it was common knowledge 🤔 other than that if you are curious about ticket type, maybe head-over our Product content first

**Comment by sleepylee** [comment]
https://nimblehq.co/compass/product/planning/
